File: seynekun/ClientApp/src/app/seynekun/models/modelo-materia-prima/materia-prima.ts
import { Producto } from "../modelo-producto/producto";

export class MateriaPrima {
  fecha: Date;
  cantidad: number;
  codigo: string;
  codigoProductor: string;
  nombreProductor: string;
  estadoMateria: string;
  tipo: string;
  productos: Producto[]
}

File: Entity/MateriaPrima.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Entity
{
    public class MateriaPrima
    {
        [Key]
        [Required(ErrorMessage = "Se requiere un código")]
        public string Codigo { get; set; }
        [Required(ErrorMessage = "Se requiere la fecha de ingreso")]
        public DateTime Fecha { get; set; }
        public string UnidadMedida { get; set; }
        [Required(ErrorMessage="Se requiere el tipo de materia prima que ingresó")]
        public string Tipo { get; set; }
        public string NombreProductor { get; set; }
        [Required(ErrorMessage = "Se requiere el código del productor")]
        public string CodigoProductor { get; set; }
        [Required(ErrorMessage = "Se requiere la cantidad ingresada de materia prima")]
        public decimal Cantidad { get; set; }
        public string EstadoMateria { get; set; }
    }
}
